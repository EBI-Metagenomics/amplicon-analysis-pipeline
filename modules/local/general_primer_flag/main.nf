
process GENERAL_PRIMER_FLAG {
    // Check for the presence of primers in general
    tag "$meta.id"
    label 'very_light'
    // TODO: uncomment container when you release fix to mpt
    // container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
    //     "https://depot.galaxyproject.org/singularity/mgnify-pipelines-toolkit:${params.mpt_version}":
    //     "biocontainers/mgnify-pipelines-toolkit:${params.mpt_version}" }"

    input:
    tuple val(meta), path(reads_merged)

    output:
    tuple val(meta), path("*general_primer_out.txt"), emit: general_primer_out

    script:
    """
    are_there_primers -i $reads_merged -s ${meta.id}_${meta.var_region} -o ./
    """
}
