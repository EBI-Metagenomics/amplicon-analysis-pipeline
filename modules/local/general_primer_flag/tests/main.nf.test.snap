{
    "general_primer_out": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "16S-V3-V4",
                    "test_16S-V3-V4_general_primer_out.txt:md5,f2160c8ffedf48068f2e1137e0a3a7e7"
                ]
            ]
        ],
        "timestamp": "2023-11-03T13:24:19.093484741"
    }
}